include/pybind11/detail/class.h

```diff
@@ -340,6 +356,32 @@ inline bool deregister_instance(instance *self, void *valptr, const type_info *t
     return ret;
 }
 
+using derived_tp_init_registry_type
+    = std::unordered_map<PyTypeObject *, int (*)(PyObject *, PyObject *, PyObject *)>;
```

Comment:
what about [initproc](https://docs.python.org/3/c-api/typeobj.html#c.initproc) as the value type?

Reply:
Much nicer, thanks! (I totally didn't think of that)